autonomy/cli/helpers/analyse.py

```diff
@@ -88,59 +94,199 @@ def run_dialogues_check(
+    ipfs_tool = IPFSTool()
+    if is_on_chain_check:
+        if not ipfs_tool.daemon.is_started():
+            raise click.ClickException("Cannot connect to the IPFS daemon.")
+        return ipfs_tool.all_pins()
+
+    return set()
+
+
+def _get_service_hash(service_id: PackageId, package_manager: PackageManagerV1) -> str:
+    """Iterate through the available packages and find the hash for the service"""
+
+    for package_id, package_hash in package_manager.dev_packages.items():
+        if (
+            package_id.package_type == PackageType.SERVICE
+            and package_id.public_id.to_any() == service_id.public_id.to_any()
+        ):
+            return package_hash
+
+    raise click.ClickException(
+        f"Service with public id `{service_id.public_id}` not found"
+    )
+
 
 def check_service_readiness(
     token_id: Optional[int],
-    service_path: Path,
+    service_id: PackageId,
     chain_type: ChainType,
+    packages_dir: Path,
 ) -> None:
     """Check deployment readiness of a service."""
 
-    try:
-        service_analyser = ServiceAnalyser(service_path=service_path)
-        if token_id is not None:
-            click.echo("Checking if the service is deployed on-chain")
-            ledger_api, _ = get_ledger_and_crypto_objects(chain_type=chain_type)
-            service_analyser.check_on_chain_state(
-                ledger_api=ledger_api,
-                chain_type=chain_type,
-                token_id=token_id,
-            )
+    package_manager = PackageManagerV1.from_dir(packages_dir=packages_dir)
+    is_on_chain_check = token_id is not None
+    ipfs_pins = _get_ipfs_pins(is_on_chain_check=is_on_chain_check)
 
-        ipfs_pins = IPFSTool().all_pins()
+    service_id = service_id.with_hash(
+        package_hash=_get_service_hash(
+            service_id=service_id,
+            package_manager=package_manager,
+        )
+    )
+    service_config = cast(
+        Service,
+        (
+            _load_from_ipfs(package_id=service_id)
+            if is_on_chain_check
+            else _load_from_local(
+                package_id=service_id, package_manager=package_manager
+            )
+        ),
+    )
 
-        click.echo("Verifying overrides")
-        service_analyser.check_required_overrides()
+    agent_id = PackageId(
+        package_type=PackageType.AGENT,
+        public_id=service_config.agent,
+    )
+    agent_config = cast(
+        AgentConfig,
+        (
+            _load_from_ipfs(package_id=agent_id)
+            if is_on_chain_check
+            else _load_from_local(package_id=agent_id, package_manager=package_manager)
+        ),
+    )
 
-        click.echo("Checking if the agent package is published")
-        service_analyser.check_agent_package_published(ipfs_pins=ipfs_pins)
+    skill_config = _get_chained_abci_skill(
+        agent_config=agent_config,
+        package_manager=package_manager,
+        is_on_chain_check=is_on_chain_check,
+    )
```

Comment:
If token ID is provided we check the components which are published on the IPFS registry else we validate the local components